### ISSUE TITLE
Shadowing variables

### ISSUE THREAD

**Issue by DoomTas3r** [endlessm/godot-block-coding#265]
### What Happened

I thought, 'How can `delta` from `every frame` be obtained without creating that variable?' I created the variable with the exact name `delta` and printed the result which was a float. The debugger displays a warning that `delta` shadows the already-declared variable.

This unlocks a feature, but it can become a problem if the types don't match

![image](https://github.com/user-attachments/assets/1f35bc57-a1d1-455c-9b11-50f6d3ac45de)

Generated script:
```
extends Node2D

var delta: float


func _ready():
    delta = 0

func _process(delta):
    print((delta))


```

### How to Reproduce

1. Create a variable named `delta`
2. Place an `every frame` block
3. Attach a `log text` block to `every frame`
4. Attach `delta` as input to `log text`
5. Run the scene

### Block Coding Plugin Version

v0.7.1

### Godot Engine Version

v4.3

### Logs or command-line output

```shell
W 0:00:02:0492   The local function parameter "delta" is shadowing an already-declared variable at line 3.
  <GDScript Error>SHADOWED_VARIABLE
  <GDScript Source>gdscript://-9223371985063508500.gd:9
```

**Comment by manuq** [comment]
Yes, very good point. For variables maybe adding them all to a `_block_coding_variables = {}` dictionary would be an option.

Then we have the same issue when defining methods: the user can currently shadow a builtin method like Node2D `rotate()` using the "define method" block.
